climada/engine/unsequa/calc_delta_climate.py

```diff
@@ -0,0 +1,435 @@
+    calc_eai_exp : bool
+        Compute eai_exp or not
+    calc_at_event : bool
+        Compute eai_exp or not
+    value_unit : str
+        Unit of the exposures value
+    exp_input_var : InputVar or Exposures
+        Exposure uncertainty variable
+    impf_input_var : InputVar if ImpactFuncSet
+        Impact function set uncertainty variable
+    haz_input_var: InputVar or Hazard
+        Hazard uncertainty variable
+    _input_var_names : tuple(str)
+        Names of the required uncertainty input variables
+        ('exp_initial_input_var', 'impf_initial_input_var', 'haz_initial_input_var',
+         'exp_final_input_var', 'impf_final_input_var', 'haz_final_input_var'')
+    _metric_names : tuple(str)
+        Names of the impact output metrics
+        ('aai_agg', 'freq_curve', 'at_event', 'eai_exp')
+    """
+
+    _input_var_names = (
+        "exp_initial_input_var",
+        "impf_initial_input_var",
+        "haz_initial_input_var",
+        "exp_final_input_var",
+        "impf_final_input_var",
+        "haz_final_input_var",
+    )
+    """Names of the required uncertainty variables"""
+
+    _metric_names = ("aai_agg", "freq_curve", "at_event", "eai_exp")
+    """Names of the cost benefit output metrics"""
+
+    def __init__(
+        self,
+        exp_initial_input_var: Union[InputVar, Exposures],
+        impf_initial_input_var: Union[InputVar, ImpactFuncSet],
+        haz_initial_input_var: Union[InputVar, Hazard],
+        exp_final_input_var: Union[InputVar, Exposures],
+        impf_final_input_var: Union[InputVar, ImpactFuncSet],
+        haz_final_input_var: Union[InputVar, Hazard],
+    ):
+        """Initialize UncCalcImpact
+
+        Sets the uncertainty input variables, the impact metric_names, and the
+        units.
+
+        Parameters
+        ----------
+        exp_initial_input_var : climada.engine.uncertainty.input_var.InputVar or climada.entity.Exposure
+            Exposure uncertainty variable or Exposure of initial state
+        impf_initital_input_var : climada.engine.uncertainty.input_var.InputVar or climada.entity.ImpactFuncSet
+            Impact function set uncertainty variable or Impact function set of initial state
+        haz_initial_input_var : climada.engine.uncertainty.input_var.InputVar or climada.hazard.Hazard
+            Hazard uncertainty variable or Hazard of initial state
+        exp_final_input_var : climada.engine.uncertainty.input_var.InputVar or climada.entity.Exposure
+            Exposure uncertainty variable or Exposure of final state
+        impf_final_input_var : climada.engine.uncertainty.input_var.InputVar or climada.entity.ImpactFuncSet
+            Impact function set uncertainty variable or Impact function set of final state
+        haz_final_input_var : climada.engine.uncertainty.input_var.InputVar or climada.hazard.Hazard
+            Hazard uncertainty variable or Hazard of final state
+
+        """
+
+        Calc.__init__(self)
```

Comment:
Not really necessary, but safer in general
```suggestion
        super().__init__(self)
```

Reply:
This actually created an error in the tests. I thus reverted it back to Calc.__init__(self).